### PR TITLE
Add postinstall script, update dependencies, update README instructions.

### PR DESCRIPTION
**Pull request by ElmoArmy**
### Description of changes proposed in this pull request:

- Added prepack script, runs build and docgen before packing repo for publishing
- Added postinstall script, runs hardhat clean, hardhat compile, and tsc after yarn install
- Added clean  script, runs hardhat clean.
- Added comments to the env template file.
- Updated dependencies and devDependencies. 
- Added instructions regarding the ENV template to the README. 

### Checks:

- [X] Have you followed the guidelines in our Contributing document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submission:

- [X] Have you successfully ran tests with your changes locally?
- [X] Have you lint your code locally prior to submission?
One lint warning not introduced by this commit,  #103 created in response.

### Changes to Existing Features:
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

**Comment by ElmoArmy** [comment]
@byterose should we also remove the dist directory within the clean script? If so...

 We could just add "rm-rf dist/ && hardhat clean" , but another way to do it could be programmatically by [overriding](https://hardhat.org/guides/create-task.html#overriding-tasks) the hardhat clean task. I propose that because it'd be easier to extend and get around cross env issues.

**Comment by byterose** [comment]
> @byterose should we also remove the dist directory within the clean script? If so...
> 
> We could just add "rm-rf dist/ && hardhat clean" , but another way to do it could be programmatically by [overriding](https://hardhat.org/guides/create-task.html#overriding-tasks) the hardhat clean task. I propose that because it'd be easier to extend and get around cross env issues.

I think it is better to add `rm -fr dist/ && hardhat clean` into the script, just to make it more transparent to the developer what the script is doing.

File: README.md [hunk 1]
```diff
@@ -9,7 +9,10 @@
 
 # Using the Yam Uma Synths SDK
 
-### **Official assets:** [File](https://github.com/yam-finance/synths-sdk/blob/master/src/assets.json) and [Raw file](https://raw.githubusercontent.com/yam-finance/synths-sdk/master/src/assets.json)
+### \*\*Official
```
Comment (byterose) [inline README.md:12]: Why do you do this?
 
 Have a look at how this is rendered - it looks like this is not what you want to achieve.

Comment (ElmoArmy) [reply README.md]: IDE sillyness. Reverted with b1da1cc

**Pull request merged**